### ISSUE TITLE
AWSFacebookSignInProvider provides no error and no completion return upon cancel or done.  AWSGoogleSignInProvider does. 

### ISSUE THREAD

**Issue by BruceBuckland** [aws/aws-mobilehub-helper-ios#13]
When we loginWithSignInProvider with the AWSFacebookSignInProvider, and the user clicks "Done" or "Cancel" we don't even get the completion routine run, and we get no error and no status.  This makes it very difficult (impossible) for an AWSSignInProviderManager to determine if there is a login or not (because it never gets control so it cannot even poll isLoggedIn.)

In the same situation with "Done" clicked, AWSGoogleSignInProvider DOES provide an error (which allows the AWSSignInProviderManager to properly manage active sessions.

In AWSFacebookSignInProvider (This may be an oversignt) at line 207 comment reads:
                                              } else if (result.isCancelled) {
                                                   // Login canceled, do nothing

But a better idea would be to tell call the completion routine with a nice error like google does (Google calls this  "User cancelled the sign-in flow." for the "Done" click, and "access_denied" for "Deny"). 

If the designer's have an authoritative source for what is logged in and what is not, the caller can use that.  But I don't see one, it doesn't make sense to poll isLoggedIn and the caller does not even get the completion routine run.  (This sort of makes it an **in**completion routine).

I think this is a bug.  It could be called an enhancement if AWSFacebookSignInProvider returned control at all.  But I suggest the best fix is to make it behave like AWSGoogleSignInProvider and return an error.

**Comment by danieldelouya** [comment]
+1 wondering about the same thing

**Comment by BruceBuckland** [comment]
@danieldelouya if it helps, I have a fix in a fork off of this repo, which changes AWSFacebookSignInProvider.m to return an error in that case.  I did not suggest my fix above however, because my fix is a bit of a kludge (I didn't define an error code, I just use an existing error and put different userInfo in the NSError so that I can handle it the same way I do for Google. It is also possible to fix it in facebook code (FBSDKLoginManager.m at line 550 handleImplicitCancelOfLogIn and at line 219.  I looked there, and did not find, a way this might be effected by setLoginBehavior).

So I just patched it, and reported it as an issue. I know it is ugly, but at least it works fine, enough to let AWSIdentityManager manage (and do identity merging and tracking of active sessions).
